### PR TITLE
[Inbound outbound] call switching UI

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Add call switching UI
image
image

# Why

Allows the user to manage multiple ongoing calls

https://skype.visualstudio.com/SPOOL/_workitems/edit/3786032
# How Tested

Sample built and tested locally

### CODE REVIEW

File: samples/CallingStateful/src/index.css [hunk 1]
```diff
@@ -1,38 +1,29 @@
...
 }
 
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
-  }
-  to {
-    transform: rotate(360deg);
-  }
+.ui-provider {
```
Comment (JamesBurnside) [inline samples/CallingStateful/src/index.css:27]: this shouldn't be needed anymore with the removal of fluent northstar

File: samples/CallingStateful/src/App.tsx [hunk 2]
```diff
@@ -42,16 +45,20 @@ function App(): JSX.Element {
...
   /**
    * Helper function to clear the old incoming Calls in the app that are no longer valid.
    * @param statefulClient
    */
   const filterEndedIncomingCalls = useCallback(
     (incomingCall: IncomingCallCommon): void => {
-      setIncomingCalls(incomingCalls.filter((call) => call.id !== incomingCall.id));
+      console.log(incomingCalls);
+      const newIncomingCalls = incomingCalls.filter((call) => call.id !== incomingCall.id);
+      console.log(newIncomingCalls);
```
Comment (JamesBurnside) [inline samples/CallingStateful/src/App.tsx:60]: remove console logs

File: samples/CallingStateful/src/App.tsx [hunk 3]
```diff
@@ -202,26 +207,48 @@ function App(): JSX.Element {
...
+              <CallScreen
+                statefulCallClient={statefulCallClient}
+                /* @conditional-compile-remove(one-to-n-calling) */ callAgent={callAgent}
+                /* @conditional-compile-remove(one-to-n-calling) */ call={call}
+              />
+            </Stack>
+          )}
+        </Stack>
+        {calls.length > 0 && (
+          <Stack style={{ width: '20%', height: '100%' }}>
```
Comment (JamesBurnside) [inline samples/CallingStateful/src/App.tsx:228]: Be careful setting width to a `%`, these often perform poorly when the screen size is shrunk. Either give a min-width or just set a fixed width

File: samples/CallingStateful/src/components/CallManager.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      <Stack>
+        {activeCall && (
+          <Stack>
+            <Text variant="large">Active Call</Text>
+            <CallItem call={activeCall} onSetResume={onSetResume} isActiveCall={true} onSetHold={onSetHold} />
+          </Stack>
+        )}
+        {
+          <Stack>
+            {calls && calls.length > 1 && <Text variant="large">Other Calls</Text>}
```
Comment (JamesBurnside) [inline samples/CallingStateful/src/components/CallManager.tsx:31]: this is brittle in the event `calls.length === 1 && call.id !== activeCall?.id` (which shouldn't ever happen but is an edge case.
Instead consider a filter first
```
const otherCalls = calls.filter(call => call.id !== activeCall?.id);
...
{otherCalls.length > 1 && <Text variant="large">Other Calls</Text>}
{otherCalls.map((call) => ... }
```

File: samples/CallingStateful/src/components/CallManager.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  useEffect(() => {
+    if (call.state === 'Connecting') {
+      setCallTitle('Connecting');
+    } else {
+      const names = call.remoteParticipants.map((participant) => participant.displayName);
+      if (names.length === 1 && names[0]) {
+        setCallTitle(names[0]);
+      } else if (names.length > 1) {
+        setCallTitle(names.join(', '));
+      } else if (names.length === 0) {
```
Comment (JamesBurnside) [inline samples/CallingStateful/src/components/CallManager.tsx:74]: nit: just `else` here to ensure we don't leave the string in the Connecting state
```suggestion
      } else {
```

**Review by JamesBurnside — APPROVED** [review]

**Review by mgamis-msft — APPROVED** [review]

**Pull request merged**